### PR TITLE
support hide dependency and codeOnly dependency

### PR DESCRIPTION
**Pull request by theonefx**
Add more property to dependency:
- hide: Wheather dependency not display in metadata API.
- codeOnly: Wheather dependency not add any gradle | maven dependency, but only attach corresponding sample code to generated project.
- dependencies: The sub dependencies of current dependency. If dependencies is set, sub dependency will auto patched to generated project.

#22

### CODE REVIEW

File: initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildWriter.java [hunk 1]
```diff
@@ -213,36 +212,53 @@ private void writeScm(IndentingWriter writer, MavenScm mavenScm) {
...
+                .filter((dep) -> filter.test(dep.getScope()))
+                .sorted(getDependencyComparator())
+                .collect(Collectors.toList());
         writeCollection(writer, candidates, this::writeDependency);
         return candidates;
     }
 
+    private boolean isNotCodeOnly(String id) {
+        io.spring.initializr.metadata.Dependency dep = metadata.getDependencies().get(id);
+        if (dep == null) {
```
Comment (steverao) [inline initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildWriter.java:252]: Should to check the logic, I saw a same other dependencies such as spring-boot-starter or spring-boot-starter-test are considered to be CodeOnly.
image

**Review by steverao — COMMENTED** [review]
In addition, you can change the `组件依赖` to `组件依赖或示例代码`.

File: initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildProjectContributor.java [hunk 2]
```diff
@@ -90,58 +95,63 @@ public void contribute(Path projectRoot) throws IOException {
...
-                                    .map(name -> "<module>" + name.getName() + "</module>")
-                                    .forEach(writer::println));
-                            writer.println("</modules>\n");
-                        });
-
-                        rewrited.append(sw);
+                // create new pom.xml file
+                super.contribute(projectRoot);
+
+                // insert <modules /> to pom.xml
```
[the remark this answers is not in the document]
Comment (theonefx) [reply initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildProjectContributor.java:117]: The comment is explaining what the following code does, but not code which i deleted.
So, this is correct.

File: initializer-generator/src/main/java/com/alibaba/initializer/controller/InitializerMetadataV2JsonMapper.java [hunk 3]
```diff
@@ -89,10 +93,49 @@ protected void architecture(ObjectNode parent, ArchitectureCapability capability
...
         }
         ArrayNode values = nodeFactory().arrayNode();
-        values.addAll(capability.getContent().stream().map(this::mapValue)
-                .collect(Collectors.toList()));
+        values.addAll(capability.getContent().stream().map(this::mapValue).collect(Collectors.toList()));
         single.set("values", values);
         parent.set(capability.getId(), single);
     }
 
+    protected void dependencies(ObjectNode parent, DependenciesCapability capability) {
```
[the remark this answers is not in the document]
Comment (theonefx) [reply initializer-generator/src/main/java/com/alibaba/initializer/controller/InitializerMetadataV2JsonMapper.java:102]: done

File: initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildWriter.java (hunk 1 above)
Comment (theonefx) [reply initializer-generator/src/main/java/com/alibaba/initializer/generation/extension/build/maven/MulitModuleMavenBuildWriter.java:252]: done

**Pull request merged**